config/rbac/kustomization.yaml

```diff
@@ -25,6 +25,7 @@ resources:
 - user-cluster-role.yaml
 - operator-role-events.yaml
 - operator-role-knative.yaml
+- operator-role-olm.yaml
```

Comment:
What is this new role for? I fail to see why it should be bound to the operator ServiceAccount.

Reply:
There is a tecton test which installs kamel and is executed by the camel-k-operator service-account. On OCP, the OLM is available hence it tries to install via the mechanism but throws permission errors as the SA has no privileges for subscriptions etc... Therefore, this role allows this test to pass.

Reply:
We aim to keep the set of permissions for the operator reduced to the minimum. Extra permissions for OLM resources are not needed by the operator, only by the account that's used to run the Kamel CLI install command.  In the Tekton test case, it seems the problem is two folds:
* The test re-use the `camel-k-operator` ServiceAccount, while it should create one dedicated to calling the Kamel CLI,
* The test runs the install command, but I doubt it's a real use case. Usually, Tekton pipelines would run integrations, not install Camel K.

They are some documentation available at https://camel.apache.org/camel-k/1.8.x/tutorials/tekton/tekton.html. That being said, I doubt this feature has ever been actually used.

Reply:
I'll set this test as problematic then and back out the olm-operator permission role.

Reply:
Sounds good 👍🏼.